src/auto-evo/mutations/CommonMutationFunctions.cs

```diff
@@ -0,0 +1,237 @@
+        OrganelleLayout<OrganelleTemplate> existingOrganelles, Direction direction, Random random)
+    {
+        var result = new OrganelleTemplate(organelle, new Hex(0, 0), 0);
+
+        // Loop through all the organelles and find an open spot to
+        // place our new organelle attached to existing organelles
+        // This almost always is over at the first iteration, so its
+        // not a huge performance hog
+        foreach (var otherOrganelle in existingOrganelles.OrderBy(_ => random.Next()))
+        {
+            // The otherOrganelle is the organelle we wish to be next to
+            // Loop its hexes and check positions next to them
+            foreach (var hex in otherOrganelle.RotatedHexes)
+            {
+                // Offset by hexes in organelle we are looking at
+                var pos = otherOrganelle.Position + hex;
+
+                foreach (int side in SideTraversalOrder(hex, direction, random))
+                {
+                    // pick a hex direction, with a slight bias towards forwards
+                    for (int radius = 1; radius <= 3; ++radius)
+                    {
+                        // Offset by hex offset multiplied by a factor to check for greater range
+                        var hexOffset = Hex.HexNeighbourOffset[(Hex.HexSide)side];
+                        hexOffset *= radius;
+                        result.Position = pos + hexOffset;
+
+                        if (organelle.HasMovementComponent)
+                        {
+                            // Face movement to move forward
+                            result.Orientation = 3;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+
+                        // Check every possible rotation value.
+                        for (int rotation = 0; rotation <= 5; ++rotation)
+                        {
+                            result.Orientation = rotation;
+
+                            if (existingOrganelles.CanPlace(result, new List<Hex>(), new List<Hex>()))
+                            {
+                                return result;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public static void AttachIslandHexes(OrganelleLayout<OrganelleTemplate> organelles)
+    {
+        var workMemory1 = new HashSet<Hex>();
+        var workMemory2 = new List<Hex>();
+        var workMemory3 = new Queue<Hex>();
+        var islandHexes = new List<Hex>();
+
+        organelles.GetIslandHexes(islandHexes, workMemory1, workMemory2, workMemory3);
+
+        // Attach islands
+        while (islandHexes.Count > 0)
+        {
+            var mainHexes = organelles.ComputeHexCache().Except(islandHexes);
```

Comment:
This should use the hex cache variant taking in an existing collection to put the result in.